Req resp bitswap

dvc94ch opened the pull request:
Closes #4 closes #5 close #6 close #7

TODO:
- [x] support canceling queries
- [x] box bitswap store

dvc94ch commented on src/behaviour.rs:
```diff
@@ -5,263 +5,368 @@
+pub struct BitswapConfig<P, S> {
+    _marker: PhantomData<P>,
+    store: S,
+    pub credit_timeout: Duration,
```
Unused, remove

rklaehn commented:
If you want to get good reviews, it might be useful to add a module level documentation about what the purpose of the thing is. Especially the query stuff. The normal bitswap have/want stuff is pretty obvious.

dvc94ch commented:
Added some comments on future work here #10, but I think it's good enough for the time being. Also added some docs, but really they're two rather simple finite state machines (one per query type) and a query manager that manages them.

dvc94ch commented:
Good enough for now, works with ipfs-embed aliasing.

rklaehn commented:
Thanks for the comment. Looks good. Agree about the boxing of the BitswapStore.
